packages/suite/src/views/wallet/trading/common/TradingForm/TradingFormInput/TradingFormInputCryptoSelect/TradingSelectAssetModal/TradingSelectAssetModal.tsx

```diff
@@ -0,0 +1,173 @@
+
+            if (!findOption) return;
+
+            if (isTradingExchangeContext(context)) {
+                context.setValue(TRADING_FORM_RECEIVE_CRYPTO_CURRENCY_SELECT, findOption, {
+                    shouldDirty: true,
+                });
+
+                context.resetSelectedOffer();
+            } else {
+                context.setValue(TRADING_FORM_CRYPTO_CURRENCY_SELECT, findOption, {
+                    shouldDirty: true,
+                });
+            }
+
+            context.setAmountLimits(undefined);
+            onModalClose();
+            dispatch(tradingActions.setModalCryptoCurrency(findOption.value));
+        },
+        [context, dispatch, onModalClose, rawOptions],
+    );
+
+    const networkCount = useNetworksCount(options);
+    const { activeTab, setActiveTab } = useNetworksTabs();
+    const { filteredOptions, setSearch, search } = useOptionsSearch(allOptions, activeTab);
+    const optionsFingerprint = useMemo(() => getFingerprint(filteredOptions), [filteredOptions]);
+
+    return (
+        <SelectAssetModal
+            data-testid={dataTestId ?? '@trading/form/select-crypto'}
+            options={filteredOptions}
+            optionsFingerprint={optionsFingerprint}
+            onSelectAsset={handleSelectChange}
+            onClose={onModalClose}
+            searchInput={
+                <SearchAsset
+                    data-testid="@trading/form/select-crypto/search-input"
+                    searchPlaceholder={translationString('TR_SELECT_NAME_OR_ADDRESS')}
+                    search={search}
+                    setSearch={setSearch}
+                />
+            }
+            noItemsAvailablePlaceholder={{
+                heading: (
+                    <Translation
+                        id="TR_TOKEN_NOT_FOUND_ON_NETWORK"
+                        values={{
+                            networkName: activeTab?.name ?? '',
+                        }}
+                    />
+                ),
+                body: <Translation id="TR_TOKEN_TRY_DIFFERENT_SEARCH_OR_SWITCH" />,
+            }}
+            filterTabs={
+                <NetworkTabs
+                    data-testid="@trading/form/select-crypto/network-tab"
+                    tabs={TOKEN_SELECT_SELECTABLE_NETWORKS}
+                    networkCount={networkCount}
+                    activeTab={activeTab}
+                    setActiveTab={setActiveTab}
+                />
+            }
+            renderOptionBalance={option =>
+                option.tokenBalance && option.contractAddress ? (
+                    <TokenBalance
+                        symbol={option.symbol}
+                        contractAddress={option.contractAddress}
+                        tokenBalance={option.tokenBalance}
+                    />
+                ) : null
+            }
+        />
+    );
+}
+
+export const TradingSelectAssetModal = memo(
+    TradingSelectAssetModalInner,
+) as typeof TradingSelectAssetModalInner;
```

Comment:
Again, extract it to standalone component and yes, there's still space for further refactoring.